docs/design/federation-phase-1.md

```diff
@@ -157,23 +168,42 @@ clusters over AWS clusters" or "spread load equally across clusters").
 
 ## Ubernetes Cluster Controller
 
-The cluster controller
-performs the following two kinds of work:
-
-1. It watches all the sub-resources that are created by Ubernetes
-   components, like a sub-RC or a sub-service. And then it creates the
-   corresponding API objects on the underlying K8S clusters.
-1. It periodically retrieves the available resources metrics from the
-   underlying K8S cluster, and updates them as object status of the
-   `cluster` API object.  An alternative design might be to run a pod
-   in each underlying cluster that reports metrics for that cluster to
-   the Ubernetes control plane.  Which approach is better remains an
-   open topic of discussion.
+The cluster controller's job is to reconcile the desired state of the
+underlying Kubernetes clusters (as recorded by Ubernetes in it's
+persistent store of desired objects) against the real Kubernetes
+clusters' states.
+
+To achieve this, it performs the following three kinds of work:
+
+1. It watches the Ubernetes persistent store for new or changed
+   desired Kubernetes resources (e.g. replication controllers or
+   services), and creates or modifies the corresponding API objects on
+   the underlying Kubernetes clusters.
+1. It watches the state of the underlying Kubernetes clusters' API
+   objects, and reacts accordingly when this is at odds with it's
+   persistent store of desired state (to cater for, e.g, the
+   possibility of some external agent deleting or modifying the
+   underlying resources).  Note that it will be necessary to be able
+   to disable this behaviour to make it possible for things like
+   Kubernetes pod autoscalers, which intentionally and automatically
+   modify Kubernetes replication controller's desired state, to
+   interoperate correctly with Ubernetes.  The details of this
+   interoperability are as yet undefined. In practise 1 and 2 are
+   implemented as a single reconciliation loop.
+1. It periodically retrieves available resource metrics from
```

Comment:
Hi, @quinton-hoole ,
from https://github.com/kubernetes/kubernetes/issues/23614#issuecomment-204148122. 
I got that , in phase 1, do we just need get the status of underlying Kubernetes clusters ,and choose a cluster randomly?
BTW, shall we use the restapi `https://kubernetes/healthz` to get the status of cluster?